### ISSUE TITLE
imshow upscale displayed image

### ISSUE THREAD

**Issue by cdcseacave** [opencv/opencv#11360]
Using the latest OpenCV on Windows 10, any image I show or any options I use for creating the window (autosize, or normal with specified size) the resulted window displays a scaled version of the image. The scale seems to correspond to the general scale set in Windows display settings. Is there a way to disable that scale, at least when manually specifying the windows size? It is very important to preserve the original resolution while developing computer vision algorithms.

**Comment by catree** [comment]
@cdcseacave 
Are you talking about this?
![image](https://user-images.githubusercontent.com/8229425/39065734-14565d04-44d3-11e8-9983-0f7c80651327.png)

> It is very important to preserve the original resolution while developing computer vision algorithms.

It should be just Windows that scales the image for better visibility when rendered I think. Your computer vision algorithms will be performed on the original image size.

Maybe you could try [this](https://answers.microsoft.com/en-us/windows/forum/windows_10-performance-winpc/adapt-scaling-for-a-specific-app/f02b4b06-58e5-4790-a2f1-306ff68021fd) but I don't think it will work.

**Comment by cdcseacave** [comment]
Yes, that is what I am talking about. Thank you for pointing me to UI approach of solving this, but I was looking for a way to solve it programmatically. Should we be able to enable/disable that scaling per window, or at least per app but from inside, not ask the user to do it? For example [IrfanView](https://www.irfanview.com/) displays the images at real resolution (scale) without installation procedure (zip binary) or any action from the user interface.

I am not saying this is an OpenCV issue, but I was hoping somebody knows a way to control this as I dind't find one. Thank you!

**Comment by catree** [comment]
Maybe it is possible to change this behavior by tweaking the code used in the Windows GUI back-end in OpenCV. This [Per-window DPI Awareness Sample](https://github.com/Microsoft/Windows-classic-samples/tree/master/Samples/DPIAwarenessPerWindow) seems to use the same back-end than in [window_w32.cpp](https://github.com/opencv/opencv/blob/3.4.1/modules/highgui/src/window_w32.cpp):
- [High DPI Scaling Improvements for Desktop Applications and “Mixed Mode” DPI Scaling in the Windows 10 Anniversary Update](https://blogs.windows.com/buildingapps/2016/10/24/high-dpi-scaling-improvements-for-desktop-applications-and-mixed-mode-dpi-scaling-in-the-windows-10-anniversary-update/)

Another option should be to rely on another library for the GUI:
- [Qt High DPI Displays](https://doc.qt.io/qt-5/highdpi.html)
- [High DPI Desktop Application Development on Windows](https://msdn.microsoft.com/en-us/library/windows/desktop/mt843498(v=vs.85).aspx)

**Comment by StefanRickli** [comment]
I ran accross this issue as well today. Documented it [here](https://stackoverflow.com/questions/54149013/opencv-imshow-window-too-large-blurred-image).

One could at least offer the programmatic solution with the call to SetProcessDpiAwareness() as a workaround if one desperately needs a Python session without Windows scaling the OpenCV windows.
